docs/fsharp/whats-new/fsharp-10.md

```diff
@@ -0,0 +1,382 @@
+
+```fsharp
+[<Fact>]
+let ``works correctly`` () =  // Correct: function with unit parameter
+    Assert.True(true)
+```
+
+**Important compatibility note:**
+
+This is a breaking change that may reveal previously silent issues in your codebase.
+The early errors prevent test discovery problems and ensure that attributes like analyzers and decorators take effect as intended.
+
+## Support for `and!` in task expressions
+
+You can now await multiple tasks concurrently using [`and!`](../language-reference/computation-expressions.md#and) in [task expressions](../language-reference/task-expressions.md).
+Using `task` is a popular way to work with asynchronous workflows in F#, especially when you need interoperability with C#.
+However, until now there was no concise way to await multiple tasks concurrently in a computation expression.
+
+Perhaps you started with code that awaited computations sequentially:
+
+```fsharp
+// Awaiting sequentially
+task {
+    let! a = fetchA()
+    let! b = fetchB()
+    return combineAB a b
+}
+```
+
+If you then wanted to change it to await them concurrently, you would typically use `Task.WhenAll`:
+
+```fsharp
+// Use explicit Task combinator to await concurrently
+task {
+    let ta = fetchA()
+    let tb = fetchB()
+    let! results = Task.WhenAll([| ta; tb |])
+    return combineAB ta.Result tb.Result
+}
+```
+
+In F# 10, you can use `and!` for a more idiomatic approach:
+
+```fsharp
+task {
+    let! a = fetchA()
+    and! b = fetchB()
+    return combineAB a b
+}
+```
+
+This combines the semantics of the concurrent version with the simplicity of the sequential version.
+
+This feature implements F# language suggestion [#1363](https://github.com/fsharp/fslang-suggestions/issues/1363).
+
+## Better trimming by default
+
+F# 10 removes a long-standing bit of friction with trimming F# assemblies.
+Trimming is the process of removing unused code from your published application to reduce its size.
+You no longer have to manually maintain an `ILLink.Substitutions.xml` file just to strip large F# metadata resource blobs (signature and optimization data that the compiler uses but your application doesn't need at runtime).
+
+When you publish with trimming enabled (`PublishTrimmed=true`), the F# build now automatically generates a substitutions file that targets these tooling-only F# resources.
+
+Previously, you had to manually maintain this file to strip the metadata.
+This added maintenance burden and was easy to forget.
+
+The result is smaller output by default, less repetitive code to maintain, and one fewer maintenance hazard.
+If you need full manual control, you can still add your own substitutions file.
+You can turn off the auto-generation with the `<DisableILLinkSubstitutions>false</DisableILLinkSubstitutions>` property.
+
+## Parallel compilation in preview
+
+An exciting update for F# users looking to reduce compilation times: the parallel compilation features are stabilizing.
```

Comment:
Somewhere it should state the limitation that this only works for builds which are not marked as deterministic (i.e. when someone uses `<Deterministic>true<..>`, it remains single-threaded